### ISSUE TITLE
Support strikethrough text style

### ISSUE THREAD

**Issue by Tyriar** [xtermjs/xterm.js#580]
I notice these aren't implemented:

- ~~Dim `\e[2m` -> `opacity: 50%`~~ (Done)
- ~~Italics `\e[3m` -> `font-style: italic`~~ (Done https://github.com/xtermjs/xterm.js/pull/1422)
- Strikethrough `\e[9m` -> `text-decoration: line-through`

**Comment by LucaT1** [comment]
Is this implemented as of today?
The css is here, but for me it's not working inside of vim, macOS

**Comment by Tyriar** [comment]
It's not implemented, what CSS are you referring to?

**Comment by LucaT1** [comment]
Sorry for the CSS part, I thought to have seen it [there](https://github.com/sourcelair/xterm.js/blob/master/dist/xterm.css#L208-L222)

Are you planning on implementing this soon or is this a low priority task?

**Comment by Tyriar** [comment]
Those are for the ones that are implemented.

This is low priority and is blocked by https://github.com/sourcelair/xterm.js/pull/756 which is reimplementing how the character attributes are stored.

**Comment by mofux** [comment]
Here is a nice script to test the different styles:

https://gist.githubusercontent.com/mxmerz/92e97cd27857a9ba787b/raw/c16c216d11c3fa6e7f0254e2d5f485164cddfb96/ANSI-Test.sh

**Comment by LucaT1** [comment]
Cool! 😍

**Comment by flybayer** [comment]
Oh man, I want this so bad. 
It looks like this is no longer blocked by anything, correct?

**Comment by felixfbecker** [comment]
Strike through would be really great. Git supports it as a diff style: `git config --global diff.color.old "red strike"`

PowerGit uses it for gone branches:

![image](https://user-images.githubusercontent.com/10532611/57560939-5da02180-7389-11e9-9306-06d8e3353b5a.png)

**Comment by Tyriar** [comment]
If anyone wants to help out with this you will want to copy how underline works in the DOM (DomRendererRowFactory.ts) and canvas renderers (TextRenderLayer.ts), for the latter a new method like `fillBottomLineAtCells` will need to be done which draws a line through the middle.

**Comment by Tyriar** [comment]
DOM and canvas renderer support is merged thanks to @silamon, webgl is still outstanding. It's a little harder there since we don't do underline yet either https://github.com/xtermjs/xterm.js/issues/2251
